source/renderer/app/components/staking/stake-pools/StakePoolsRanking.js

```diff
@@ -0,0 +1,316 @@
+    description: 'Ranking description.',
+  },
+  rankingOneWalletEnd: {
+    id: 'staking.stakePools.rankingOneWalletEnd',
+    defaultMessage: '!!!wallet.',
+    description: 'One wallet description after dropdown.',
+  },
+  rankingOneWalletStart: {
+    id: 'staking.stakePools.rankingOneWalletStart',
+    defaultMessage:
+      '!!!Stake pools are currently ranked based on the amount in',
+    description: 'One wallet description before dropdown.',
+  },
+  rankingSelectWallet: {
+    id: 'staking.stakePools.rankingSelectWallet',
+    defaultMessage: '!!!select a wallet',
+    description: 'Select wallet item of dropdown.',
+  },
+  rankingSelectWalletEnd: {
+    id: 'staking.stakePools.rankingSelectWalletEnd',
+    defaultMessage: '!!!to set the amount you intend to delegate.',
+    description: 'Select wallet description after dropdown.',
+  },
+  rankingSelectWalletStart: {
+    id: 'staking.stakePools.rankingSelectWalletStart',
+    defaultMessage: '!!!Or',
+    description: 'Select wallet description before dropdown.',
+  },
+  rankingExtraTooltip: {
+    id: 'staking.stakePools.rankingExtraTooltip',
+    defaultMessage: '!!!Circulating supply',
+    description: 'Circulating supply slider tooltip.',
+  },
+  rankingMaxTooltip: {
+    id: 'staking.stakePools.rankingMaxTooltip',
+    defaultMessage: '!!!Saturation point',
+    description: 'Saturation point slider tooltip.',
+  },
+  rankingMinTooltip: {
+    id: 'staking.stakePools.rankingMinTooltip',
+    defaultMessage: '!!!Minimum ADA required for staking',
+    description: 'Minimum ADA required for staking slider tooltip.',
+  },
+  actionLearnMore: {
+    id: 'staking.stakePools.learnMore',
+    defaultMessage: '!!!Learn more',
+    description: 'Learn more action of ranking panel.',
+  },
+});
+
+const learnMoreUrlMap = {
+  'en-US': 'https://iohk.zendesk.com/hc/en-us',
+  'ja-JP': 'https://iohk.zendesk.com/hc/ja',
+};
+
+const walletSelectorLanguageMap = {
+  'en-US': 'en-selector',
+  'ja-JP': 'ja-selector',
+};
+
+type Props = {
+  wallets: Array<Wallet>,
+  currentLocale: string,
+  onOpenExternalLink: Function,
+  onRank: Function,
+  isLoading: boolean,
+};
+
+type State = {
+  selectedWalletId: ?string,
+  sliderValue: number,
+};
+
+const OUT_OF_RANGE_MAX_AMOUNT = new BigNumber('11000000000');
+const MIN_AMOUNT = new BigNumber('10');
+const MAX_AMOUNT = new BigNumber('44000000');
+
+@observer
+export default class StakePoolsRanking extends Component<Props, State> {
+  static contextTypes = {
+    intl: intlShape.isRequired,
+  };
+
+  static defaultProps = {
+    wallets: [],
+  };
+
+  state = {
+    selectedWalletId: null,
+    sliderValue: MIN_AMOUNT.toNumber(),
+  };
+
+  componentDidMount() {
+    this.onSelectedWalletChange('0');
+  }
+
+  getAllAvailableAmount = () => {
+    const { wallets } = this.props;
+
+    if (wallets.length) {
+      return wallets
+        .map((w: Wallet) => w.availableAmount)
```

Comment:
@yakovkaravelov `w` is not good declaration. Please declare a single wallet with `wallet`

Reply:
@tomothespian this variable `w` is only for `iterator` variable of `map` function which is in strict scope.
That's why I used `w` for simplicity. Other variables should be named in full form obviously as you mentioned.

Reply:
@yakovkaravelov ok, just mentioned because maybe will be more declarative and easier to read.